### PR TITLE
Reorganize `develop` docs sidebar

### PR DESCRIPTION
**Pull request by antonpirker**
## DESCRIBE YOUR PR
Inspired by https://github.com/getsentry/sentry-docs/pull/11619 and with the input of @markstory I updated the organization of content in our develop docs. 

No content was changed/added, I moved pages around and added a few pages that group existing pages in a better way. 

### Preview:
https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/getting-started/

### TODOs:
- [ ] Get some feedback
- [ ] Add redirects when sure that content structure will not change anymore. 

**(updated)** Here a before/after comparison of the develop docs sidebar ([miro link](https://miro.com/app/board/uXjVLJYf6Gw=/)):

Screenshot 2024-11-16 at 10 03 13

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: develop-docs/architecture/index.mdx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Application
+title: Architecture
```
Comment (lizokm) [inline develop-docs/architecture/index.mdx:2]:
```suggestion
title: Application Architecture
```
I think it's easier to quickly grasp "Application/App Architecture" than just "Architecture" by itself.

File: develop-docs/ingestion/index.mdx [hunk 2]
```diff
@@ -1,12 +1,13 @@
 ---
-title: Relay Development
-sidebar_order: 70
+title: Ingestion
+sidebar_order: 90
 ---
 
-Relay is a service for event filtering, rate-limiting and processing. It can act as:
+Ingestion is done by [Sentry Relay](https://github.com/getsentry/relay). Relay is a service for event filtering, rate-limiting and processing.
+
```
Comment (lizokm) [inline develop-docs/ingestion/index.mdx:7]:
```suggestion
Ingestion is done by [Sentry Relay](https://github.com/getsentry/relay), a service for event filtering, rate-limiting, and processing.

```

**Review by lizokm — APPROVED** [review]
Thanks for making these updates! They make sense to me :)

**Comment by antonpirker** [comment]
This is the script to check if all the sidebar items are still on the same url, or redirect somewhere that is not a 404:

```python
import requests


def main():
    urls = [
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/getting-started/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/getting-started/philosophy/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/getting-started/inclusive-language/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/environment/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/environment/ports/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/environment/u2f/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/environment/pycharm/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/commit-messages/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/code-review/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/documentation/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/workflow/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/testing/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/continuous-integration/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/ngrok/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/devservices/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/rust/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/development/analytics/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/architecture/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/dynamic-sampling/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/dynamic-sampling/the-big-picture/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/dynamic-sampling/fidelity-and-biases/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/dynamic-sampling/architecture/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/dynamic-sampling/outcomes/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/feedback-architecture/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/control-silo/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/cross-region-replication/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/cross-region-rpc/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/config/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/application/sentry-vs-getsentry/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/frontend/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/frontend/development-server/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/frontend/design-tenets/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/frontend/component-library/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/basics/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/design/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/concepts/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/public/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/checklist/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/serializers/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/api/design/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/development-server/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/python-dependencies/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/database-migrations/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/feature-flags/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/feature-flags/flagpole/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/feature-flags/options-backed-features/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/options/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/transaction-clustering/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/grouping/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/outboxes/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/issue-platform/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/issue-platform-detectors/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/queue/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/email/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/kafka/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/metrics/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/nodestore/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/digests/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/quotas/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/tsdb/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/pii/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/pii/methods/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/pii/types/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/pii/selectors/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/buffers/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/translations/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/backend/ab-testing/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/sdk/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/sdk/expected-features/rate-limiting/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/relay/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/relay/projectconfig-versioning/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/relay/transaction-span-ratelimits/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/services/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/services/chartcuterie/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/integrations/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/integrations/azuredevops/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/self-hosted/",
        "https://develop-docs-git-antonpirker-develop-docs-sidebar.sentry.dev/self-hosted/geolocation/",
    ]

    for url in urls:
        response = requests.get(url)
        if response.status_code >= 400:
            print(f"{url} - ~~~{response.status_code}~~~")
        else:
            print(f"{url} - OK")


if __name__ == "__main__":
    main()

```

**Pull request merged**